mediation_adapter_mopub/src/main/java/com/hmscl/huawei/ads/mediation_adapter_mopub/banner.kt

```diff
@@ -46,129 +48,184 @@ class banner : BaseAd() {
                     MoPubErrorCode.NETWORK_NO_FILL
-            )
+                )
+                mLoadListener?.onAdLoadFailed(MoPubErrorCode.NETWORK_NO_FILL)
+                return
+            }
 
-            mLoadListener?.onAdLoadFailed(MoPubErrorCode.NETWORK_NO_FILL)
-            return
-        }
+            val builder = AdParam.Builder()
+            builder.setRequestOrigin("MoPub")
+
+            if(!extras.containsKey(CONTENT_URL_KEY)){
+                Log.e("TAG", "Banner - load() - adData.extras is not contain contentUrl")
+            }
 
-        val builder = AdParam.Builder()
-        builder.setRequestOrigin("MoPub")
+            val contentUrl = extras[CONTENT_URL_KEY]
 
-        val contentUrl = extras[CONTENT_URL_KEY]
-        if (!TextUtils.isEmpty(contentUrl)) {
-            builder.setTargetingContentUrl(contentUrl)
-        }
+            if (!TextUtils.isEmpty(contentUrl)) {
+                builder.setTargetingContentUrl(contentUrl)
+                Log.d("TAG", "Banner - load() - adData.extras => {contentUrl = $contentUrl}")
 
-        /**
-         * Prepare Child-protection keys
-         */
-        val requestConfigurationBuilder = prepareBuilderViaExtras(extras)
-        val requestConfiguration = requestConfigurationBuilder.build()
-        HwAds.setRequestOptions(requestConfiguration)
+            }else{
+                Log.e("TAG", "Banner - load() - adData.extras => contentUrl key is empty")
+            }
+
+            /**
+             * Prepare Child-protection keys
+             */
+            val requestConfigurationBuilder = prepareBuilderViaExtras(extras)
+            val requestConfiguration = requestConfigurationBuilder.build()
+            HwAds.setRequestOptions(requestConfiguration)
+
+            val adRequest = builder.build()
+            mHuaweiAdView.loadAd(adRequest)
+            MoPubLog.log(adNetworkId, AdapterLogEvent.LOAD_ATTEMPTED, ADAPTER_NAME)
+            Log.d("TAG", "Banner - load() - adapter attempting to load ad")
+
+        } catch (e: Exception) {
+            val stacktrace =
+                StringWriter().also { e.printStackTrace(PrintWriter(it)) }.toString().trim()
+            Log.e("TAG", "Banner - loadAd() - Request Banner Ad Failed: $stacktrace")
+            mHuaweiAdView.adListener.onAdFailed(AdParam.ErrorCode.INNER)
+        }
 
-        val adRequest = builder.build()
-        mHuaweiAdView.loadAd(adRequest)
-        MoPubLog.log(adNetworkId, AdapterLogEvent.LOAD_ATTEMPTED, ADAPTER_NAME)
     }
 
     override fun getAdView(): View? {
+        Log.d("TAG", "Banner - getAdView()")
         return mHuaweiAdView
     }
 
     override fun onInvalidate() {
+        Log.d("TAG", "Banner - onInvalidate()")
         Views.removeFromParent(mHuaweiAdView)
         mHuaweiAdView.adListener = null
         mHuaweiAdView.destroy()
     }
 
     override fun getLifecycleListener(): LifecycleListener? {
+        Log.d("TAG", "Banner - getLifecycleListener()")
```

Comment:
Why does this method return always null?

Reply:
Related issue will be fixed in the next optimization.